File: daily-katas/weather-and-soccer-data/max_spread.rb
#!/usr/bin/env ruby

filename = 'weather.dat'

track = {spread: -1, day: 0};
File.readlines(filename).each do |line|
  next unless (line =~ /^\s+\d+/)
  (day, max, min) = line.gsub(/^\s+/, '').split(/\s+/)
  spread = max.to_i - min.to_i
  if spread > track[:spread]
    track = {spread: spread, day: day}
  end
end
puts "Day #{track[:day]} has spread of #{track[:spread]}"

File: daily-katas/weather-and-soccer-data/max_spread.pl
#!/usr/bin/perl

use strict;

my $filename = 'weather.dat';
open(IN, $filename);
my %track = (spread => 0, day => 0);
while (<IN>) {
    next unless (/^\s+\d+/);
    s/^\s+//;
    my ($day, $max, $min, @rest) = split(/\s+/);
    my $today_spread = $max - $min;
    if ($today_spread > $track{spread}) {
        %track = (spread => $today_spread, day => $day);
    }
}
print "Day $track{day} has spread of $track{spread}\n";
